realm/realm-library/src/main/java/io/realm/internal/TableView.java

```diff
@@ -68,6 +68,28 @@ protected TableView(Context context, Table parent, long nativePtr, TableQuery qu
         this.query = query;
     }
 
+    /**
+     * Creates a copy of already created Java TableView Object.
+     * The method is not supposed to be called by the user of the db. The method is for internal use only.
+     *
+     * @param tableView An existing TableView
+     * @param nativeViewPtr A pointer to a clone of the existing TableView.
+     */
+    protected TableView(TableView tableView, long nativeViewPtr) {
+        long nativeQueryPtr = nativeWhere(nativeViewPtr);
+        TableQuery q;
+        try {
+            q = new TableQuery(tableView.context, tableView.parent, nativeQueryPtr, this);
+        } catch (RuntimeException e) {
```

Comment:
What errors is this catching? If the native code threw any exception we should close the resource there.